src/app/worker-access.ts

```diff
@@ -0,0 +1,16 @@
+import { wrap } from "comlink";
+import type { WorkerWrapper } from "@common/worker-wrapper";
+
+export class WorkerAccess {
+  private static worker: WorkerWrapper | undefined;
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() {}
+
+  static getWorker(): WorkerWrapper {
+    if (!this.worker) {
+      this.worker = wrap(new Worker("worker.js"));
+    }
+    return this.worker;
+  }
+}
```

Comment:
Honestly, singleton pattern JavaScript/TypeScript is so uncommon! I think I would prefer just a function:

```suggestion
export function spawnNewWorker(): Remote<WorkerWrapper> {
  return wrap(new Worker("worker.js"))
}
```

Create the wrapped worker instance in `src/app/main.ts` and then explicitly _pass_ in an instance of a `Remote<WorkerWrapper>` to everything that needs it.

---

I have a rant about Singleton patterns, but I'll save that for a later time if you want to hear it lol. If we want to _ensure_ that there's only one Worker ever instantiated by the application, there's a better way to do it in JavaScript. But I think for now, I think even _that_ is overkill.

Reply:
I remember having this conversation before...